api/taskcluster/webhook.go

```diff
@@ -431,13 +431,37 @@ func (api apiImpl) GetTaskGroupInfo(rootURL string, groupID string) (*TaskGroupI
 	return &group, nil
 }
 
-func (api apiImpl) ListCheckRuns(owner string, repo string, checkSuiteID int64) (*github.ListCheckRunsResults, *github.Response, error) {
+func (api apiImpl) ListCheckRuns(owner string, repo string, checkSuiteID int64) ([]*github.CheckRun, error) {
+	var runs []*github.CheckRun
 	options := github.ListCheckRunsOptions{
 		ListOptions: github.ListOptions {
 			PerPage: 100,
 		},
 	}
-	return api.ghClient.Checks.ListCheckRunsCheckSuite(api.ctx, owner, repo, checkSuiteID, &options)
+
+	// As a safety-check, we will not do more than 5 iterations (at 100
+	// check runs per page, this gives us a 500 run upper limit).
+	hitEnd := false
+	for i := 0; i < 5; i++ {
+		result, response, err := api.ghClient.Checks.ListCheckRunsCheckSuite(api.ctx, owner, repo, checkSuiteID, &options)
+		if err != nil {
+			return runs, err
+		}
+
+		runs = append(runs, result.CheckRuns...)
+
+		// Setup for the next call.
+		if response.NextPage >= response.LastPage {
+			hitEnd = true
```

Comment:
You could simply return runs here and you won't need `hitEnd`.

Reply:
Lol, duh, thanks 🤣